### PR TITLE
[14.0][IMP] cetmix_tower_server: Update record deletion flow

### PR DESCRIPTION
**Pull request by dessanhemrayev**
Ensuring complete deletion of data in the Tower and maintaining data integrity by cascading deletion of relevant data, if necessary.

- Delete flight plan -> Delete related lines
- Delete flight plan line -> Delete related line actions
- Delete related line actions -> Delete relate variable values
- Delete flight plan log -> Delete related command logs
- Delete server -> Delete related files, secret_ids  and variable values

Task: 4048

## Release Notes

- **New Features**
  - Implemented cascading deletions for various models, ensuring that related records are automatically removed when a parent record is deleted, enhancing data integrity.

- **Bug Fixes**
  - Improved referential integrity across multiple models by updating deletion behaviors.

- **Tests**
  - Added new tests to verify the correct deletion of associated records when a flight plan or server is deleted, ensuring the integrity of the deletion process.

### CODE REVIEW

File: cetmix_tower_server/tests/test_server.py [hunk 1]
```diff
@@ -299,3 +299,100 @@ def test_server_archived_unarchived(self):
...
+            msg="Secret should be deleted when server is deleted",
+        )
+        self.assertFalse(
+            self.VariableValue.search([("id", "=", variable_value_id)]),
+            msg="Variable Value should be deleted when server is deleted",
+        )
+        self.assertFalse(
+            self.PlanLog.search([("id", "=", plan_log_id)]),
+            msg="Plan Log should be deleted when server is deleted",
+        )
```
[the remark this answers is not in the document]
Comment (dessanhemrayev) [reply cetmix_tower_server/tests/test_server.py]: good point, thanks)

File: cetmix_tower_server/models/cx_tower_server.py [hunk 2]
```diff
@@ -1589,3 +1589,11 @@ def action_open_files(self):
...
         action["context"] = context
         return action
+
+    def unlink(self):
+        for server in self:
+            server.file_ids.sudo().unlink()  # Deletes all related files
+            server.secret_ids.unlink()  # Deletes all related secrets
+            server.variable_value_ids.unlink()  # Deletes all related variable values
+            server.plan_log_ids.unlink()  # Deletes all related flight plan logs
+        return super().unlink()
```
[the remark this answers is not in the document]
Comment (dessanhemrayev) [reply cetmix_tower_server/models/cx_tower_server.py]: ok, but in this project there is no such method as _close_connections

[the remark this answers is not in the document]
Comment (dessanhemrayev) [reply cetmix_tower_server/models/cx_tower_server.py]: good, thanks

**Review by Aldeigja — APPROVED** [review]
Functional LGTM

**Review by ivs-cetmix — CHANGES_REQUESTED** [review]
Why not to use `ondelete="cascade"` instead of deleting all related records manually?

**Review by Aldeigja — APPROVED** [review]
Functional LGTM

File: cetmix_tower_server/models/cx_tower_file.py [hunk 3]
```diff
@@ -103,7 +103,9 @@ class CxTowerFile(models.Model):
         "Default value if no error happened is 'ok'.\n"
         "Otherwise there will be a server error message logged."
     )
-    server_id = fields.Many2one(comodel_name="cx.tower.server", required=True)
+    server_id = fields.Many2one(
+        comodel_name="cx.tower.server", required=True, ondelete="cascade"
+    )
```
[the remark this answers is not in the document]
Comment (ivs-cetmix) [reply cetmix_tower_server/models/cx_tower_file.py:108]: _run_auto_pull_files function doesn't depend on servers. It checks files only:

```python
@api.model
    def _run_auto_pull_files(self):
        """
        Run auto sync files
        """
        now = fields.Datetime.now()
        files = self.search(
            [
                ("source", "=", "server"),
                ("auto_sync", "=", True),
                ("sync_date_next", "<=", now),
            ]
        )
        files.download(raise_error=False)
```

**Review by ivs-cetmix — APPROVED** [review]
Code review LGTM

**Comment by CetmixGitDrone** [comment]
This PR has the `approved` label and has been created more than 5 days ago. It should therefore be ready to merge by a maintainer (or a PSC member if the concerned addon has no declared maintainer). 🤖

**Comment by CetmixGitDrone** [comment]
What a great day to merge this nice PR. Let's do it!
Prepared branch [14.0-dev-ocabot-merge-pr-151-by-ivs-cetmix-bump-nobump](https://github.com/cetmix/cetmix-tower/commits/14.0-dev-ocabot-merge-pr-151-by-ivs-cetmix-bump-nobump), awaiting test results.

**Comment by CetmixGitDrone** [comment]
Congratulations, your PR was merged at d4b23c16cf318d4c79176bc79fa5e717b4cbcc67. Thanks a lot for contributing to cetmix. ❤️

**Pull request merged**